aspnetcore/grpc/performance.md

```diff
@@ -264,3 +264,15 @@ The preceding code:
 * Attempts to get an array from `ByteString.Memory` with <xref:System.Runtime.InteropServices.MemoryMarshal.TryGetArray%2A?displayProperty=nameWithType>.
 * Uses the `ArraySegment<byte>` if it was successfully retrieved. The segment has a reference to the array, offset and count.
 * Otherwise, falls back to allocating a new array with `ByteString.ToByteArray()`.
+
+### gRPC services and large binary payloads
+
+gRPC and Protobuf can send and receive large binary payloads. Although binary Protobuf is more efficient than text-based JSON at serializing binary payloads, there are still important performance characteristics to keep in mind when working with large binary payloads.
+
+gRPC is a message based RPC framework. That means that the entire message is loaded into memory before gRPC can send it, and the entire message is deserialized into memory when received. Messages with large binary payloads can allocate byte arrays on the [large object heap](/dotnet/standard/garbage-collection/large-object-heap). Large allocations impact server performance and scalability.
+
+Advice for creating high-performance applications with large binary payloads:
+
+* **Avoid** very large binary payloads in gRPC messages.
```

Comment:
```suggestion
* **Avoid** large binary payloads in gRPC messages.
```

I'm not sure if it says anywhere what size a typical "large" binary payload is. This could state a typical value ...

> \*\*Avoid\*\* large binary payloads in gRPC messages, such as payloads that exceed approximately XXXXX MB.

Reply:
How about greater than 20% available memory? Hard to put a number on it when VMs memory size varies by such a large factor